jujugui/static/gui/src/app/app.js

```diff
@@ -2705,12 +2699,12 @@ YUI.add('juju-gui', function(Y) {
         // We don't always get a new username and password when switching
         // environments; only set new credentials if we've actually gotten them.
         // The GUI will automatically log in when we switch.
-        this.env.setCredentials({
+        this.user.controller = {
```

Comment:
I know this follows the original pattern but won't this clear the macaroons if any were set?